biz.aQute.bnd.exporters/src/aQute/bnd/exporter/feature/FeatureExporter.java

```diff
@@ -0,0 +1,259 @@
+		// .newConfigurationBuilder("pid")
+		// .build());
+		//
+		// featureBuilder.addConfigurations(Features.getBuilderFactory()
+		// .newConfigurationBuilder("factoryPid", "name")
+		// .build());
+
+		Feature f = featureBuilder.build();
+		String json = Utils.toJson(f);
+		Files.writeString(jsonPath.toPath(), json);
+		Resource jsonResource = new FileResource(jsonPath);
+	}
+
+	private Optional<String> version() {
+
+		return Optional.ofNullable(project.getBundleVersion());
+	}
+
+	private Optional<String> decription() {
+
+		return Optional.ofNullable(project.getBundleDescription());
+	}
+
+	private Optional<String> license() {
+
+		return Optional.ofNullable(project.get(Constants.BUNDLE_LICENSE));
+	}
+
+	private Optional<String> title() {
+
+		return Optional.ofNullable(project.getBundleName());
+	}
+
+	private Optional<String> vendor() {
+
+		return Optional.ofNullable(project.getBundleVendor());
+	}
+
+	private static String featureName(Project project) {
+
+		String pName = project.getName();
+		if ("bnd.bnd".equals(pName)) {
+			return "feature.json";
+		}
+		return "feature-" + pName.replace(".bndrun", ".json");
+	}
+
+	Optional<List<FeatureConfiguration>> featureConfigurations() {
+
+		return Optional.empty();
+
+	}
+
+	Optional<List<FeatureBundle>> featureBundles() {
+
+		try {
+			return Optional.ofNullable(project.getRunbundles()
+				.stream()
+				.map(Utils::toFeatureBundle)
+				.collect(Collectors.toList()));
+		} catch (Exception e) {
+
+			e.printStackTrace();
+			throw new RuntimeException(e);
+		}
+
+	}
+
+	Optional<List<Capability>> capabilities() {
+
+		Parameters parameters = project.getProvideCapability();
+		List<Capability> list = CapReqBuilder.getCapabilitiesFrom(parameters);
+		return Optional.ofNullable(list);
+
+	}
+
+	// Optional<List<MatchingRequirement>> requirement() {
+	//
+	// Parameters parameters = project.getRequireCapability();
+	// List<MatchingRequirement> list =
+	// CapReqBuilder.getRequirementsFrom(parameters)
+	// .stream()
+	// .map(MRI::new)
+	// .collect(Collectors.toList());
+	//
+	// return Optional.ofNullable(list);
+	// }
+
+	Optional<Map<String, String>> variables() {
+
+		Map<String, String> vars = project.getProperties()
+			.entrySet()
+			.stream()
+			.filter(isVariable())
+			.collect(Collectors.toMap(e -> e.getKey()
+				.toString(),
+				e -> e == null ? ""
+					: e.getValue()
+						.toString()));
+		return Optional.ofNullable(vars);
+	}
+
+	private Predicate<? super Entry<Object, Object>> isVariable() {
+
+		return e -> {
+			if (e.getKey() == null) {
+				return false;
+			}
+			if (e.getKey()
+				.toString()
+				.startsWith("-")) {
+				return false;
+			}
+			if (e.getKey()
+				.toString()
+				.startsWith("Bundle-")) {
+				return false;
+			}
+			if (e.getKey()
+				.toString()
+				.startsWith("basedir")) {
+				return false;
+			}
+			if (e.getKey()
+				.toString()
+				.startsWith("Provide-Capability")) {
```

Comment:
Seems rather arbitrary? Why worry about variables you think are not variables? Not sure I understand why you want to filter them? I could even see some use for accessing instructions & headers for documentation purposes?